searx/settings_loader.py

```diff
@@ -30,29 +31,32 @@ def load_yaml(file_name):
 
 
 def get_default_settings_path():
-    return check_settings_yml(join(searx_dir, 'settings.yml'))
+    return existing_filename_or_none(join(searx_dir, 'settings.yml'))
```

Comment:
Do we really need a function ``existing_filename_or_none`` for such a trivial task?

BTW: I would change `searx_dir` in `/share/SearXNG/searx/settings_loader.py` to an `pathlib.Path` type.

    return join(

Reply:
It makes the code of `get_user_settings_path` super easy to read.

with:
```python
    if 'SEARXNG_SETTINGS_PATH' in environ:
        return existing_filename_or_none(environ['SEARXNG_SETTINGS_PATH'])
```

without:
```python
    if 'SEARXNG_SETTINGS_PATH' in environ:
        return environ['SEARXNG_SETTINGS_PATH'] if isfile(environ['SEARXNG_SETTINGS_PATH']) else None
```